plutus-scb-client/src/View.purs

```diff
@@ -0,0 +1,235 @@
+import Chain.View (chainView)
+import Data.Array as Array
+import Data.Json.JsonMap (JsonMap, _JsonMap)
+import Data.Json.JsonUUID (JsonUUID, _JsonUUID)
+import Data.Lens (to, traversed, view)
+import Data.Lens.Extra (toArrayOf)
+import Data.Map as Map
+import Data.Maybe (Maybe(..))
+import Data.Newtype (unwrap)
+import Data.RawJson (_RawJson)
+import Data.Tuple (Tuple(..))
+import Data.Tuple.Nested (type (/\), (/\))
+import Data.UUID as UUID
+import Effect.Aff.Class (class MonadAff)
+import Halogen.HTML (ClassName(..), ComponentHTML, HTML, b_, code_, div, div_, h2_, h3_, span, span_, text)
+import Halogen.HTML.Properties (class_, classes)
+import Icons (Icon(..), icon)
+import Ledger.Crypto (PubKeyHash)
+import Ledger.Index (UtxoIndex)
+import Ledger.Tx (Tx)
+import Ledger.TxId (TxId)
+import Network.RemoteData (RemoteData(..))
+import Plutus.SCB.Events (ChainEvent(..))
+import Plutus.SCB.Events.Contract (RequestEvent, ResponseEvent)
+import Plutus.SCB.Events.Node (NodeEvent(..))
+import Plutus.SCB.Events.User (UserEvent(..))
+import Plutus.SCB.Events.Wallet (WalletEvent)
+import Plutus.SCB.Types (ActiveContract(..), ActiveContractState(..), Contract(..))
+import Plutus.SCB.Webserver.Types (FullReport(..))
+import Prelude (class Eq, class Show, otherwise, show, ($), (+), (<$>), (<<<), (<>), (==))
+import Types (HAction(..), State(State), _hooks, _partiallyDecodedResponse)
+import Wallet.Emulator.Wallet (Wallet)
+import Wallet.Rollup.Types (AnnotatedTx)
+
+render ::
+  forall m slots.
+  MonadAff m =>
+  State -> ComponentHTML HAction slots m
+render (State { chainState, fullReport }) =
+  div
+    [ class_ $ ClassName "main-frame" ]
+    [ container_
+        [ div_
+            $ case fullReport of
+                Success report -> [ fullReportPane chainState report ]
+                Failure error -> [ ajaxErrorPane error ]
+                Loading -> [ icon Spinner ]
+                NotAsked -> [ icon Spinner ]
+        ]
+    ]
+
+fullReportPane :: forall p. Chain.State -> FullReport -> HTML p HAction
+fullReportPane chainState fullReport@(FullReport { events, latestContractStatus, transactionMap, utxoIndex, annotatedBlockchain, walletMap }) =
+  row_
+    [ col10_ [ contractStatusPane latestContractStatus ]
+    , col12_ [ ChainAction <<< Just <$> annotatedBlockchainPane chainState walletMap annotatedBlockchain ]
+    , col12_ [ eventsPane events ]
+    , col8_ [ transactionPane transactionMap ]
+    , col8_ [ utxoIndexPane utxoIndex ]
+    ]
+
+contractStatusPane :: forall p i. JsonMap JsonUUID ActiveContractState -> HTML p i
+contractStatusPane latestContractStatus =
+  card_
+    [ cardHeader_
+        [ h2_ [ text "Active Contracts" ]
+        ]
+    , cardBody_
+        [ div_
+            ( ( \(Tuple k v) ->
+                  row_
+                    [ col3_ [ h3_ [ text $ view (_JsonUUID <<< to UUID.toString) k ] ]
+                    , col9_ [ text $ view (_partiallyDecodedResponse <<< _hooks <<< _RawJson) v ]
```

Comment:
could the json be formatted? `{"tx":[],"guess":["guess"],"own-pubkey":"NotWaitingForPubKey","utxo-at":[],"address":[],"tx-confirmation":[],"slot":null,"lock":["lock"]}` looks pretty meh

Reply:
I was going to make it much richer in a later commit, but you're right. Let's at least do this for now:

<img width="641" alt="Screenshot 2020-04-23 at 16 01 28" src="https://user-images.githubusercontent.com/380756/80114713-c9b86d00-857b-11ea-94ba-8bc989fe6be7.png">